test/release/examples/primers/forallLoops.chpl

```diff
@@ -0,0 +1,305 @@
+sync, single, and atomic variables.
+*/
+
+var outerIntVariable = 1;
+proc updateOuterVariable() {
+  outerIntVariable += 1;  // always refers to the outer variable
+}
+var outerAtomicVariable: atomic int;
+
+forall i in 1..n {
+  if i == 1 then
+    updateOuterVariable();     // beware of potential for data races
+
+  // the shadow variable always contains the value as of loop start
+  writeln("shadow outerIntVariable is: ", outerIntVariable);
+
+  D[i] += 0.5;                 // beware of potential for data races
+
+  outerAtomicVariable.add(3);  // ok: concurrent updates are atomic
+}
+
+writeln();
+writeln("After a loop with default intents, D is:");
+writeln(D);
+writeln("outerIntVariable is: ", outerIntVariable);
+writeln("outerAtomicVariable is: ", outerAtomicVariable.read());
+writeln();
+
+/*
+The task intents ``in``, ``const in``, ``ref``, ``const ref``,
+and ``reduce`` can be specified explicitly using a ``with`` clause.
+
+An ``in`` or ``const in`` intent creates a copy of the outer variable
+for each task. A ``ref`` or ``const ref`` makes the
+shadow variable an aliass for the outer variable.
+*/
+
+var outerRealVariable = 1.0;
+
+forall i in 1..n with (in outerIntVariable,
+                       ref outerRealVariable) {
+  outerIntVariable += 1; // a per-task copy, never accessed concurrently
+
+  if i == 1 then
+    outerRealVariable *= 2;  // beware of potential for data races
+}
+
+writeln("After a loop with explicit intents:");
+writeln("outerIntVariable is: ", outerIntVariable);
+writeln("outerRealVariable is: ", outerRealVariable);
+writeln();
+
+/*
+A reduce intent can be used to compute reductions.
+The values of each reduce-intent shadow variable at the end of its task
+is combined onto its outer variable according to the specified reduction
+operation. Within loop body, the shadow variable represents the
+accumulation state produced by this task so far, starting from
+the reduction identity value at task startup. Values can be
+combined onto this accumulation state using the reduction-specific
+operation or the ``reduce=`` operator.
+*/
+
+var outerMaxVariable = 0;
+// outerIntVariable's value before the loop will be included in the sum
+
+forall i in 1..n with (+ reduce outerIntVariable,
+                       max reduce outerMaxVariable) {
+  outerIntVariable += i;
+  if i % 2 == 0 then
+    outerMaxVariable reduce= i;
+
+  // The loop body can contain other code
+  // regardless of reduce-related operations.
+}
+
+writeln("After a loop with reduce intents:");
+writeln("outerIntVariable = ", outerIntVariable);
+writeln("outerMaxVariable = ", outerMaxVariable);
+writeln();
+
+/*
+A with-clause can be used in a similar fashion with any flavor
+of forall loop.
+
+Task-Private Variables
+----------------------
+
+A task-private variable is similar to an in-intent or ref-intent
+shadow variable in that it is initialized at the beginning of its
+task, shared by all loop iterations executed by the task, and
```

Comment:
"shared by all loop iterations executed by the task" seems like a weird phrase to me.  There's no guarantee that a task will include a loop...  What about something like "lasts the lifetime of the task" instead?